src/main/kotlin/com/github/mitchellolsthoorn/testgenie/services/CoverageVisualisationService.kt

```diff
@@ -0,0 +1,85 @@
+package com.github.mitchellolsthoorn.testgenie.services
+
+import com.github.mitchellolsthoorn.testgenie.coverage.TestGenieCoverageRenderer
+import com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsService
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diff.DiffColors
+import com.intellij.openapi.editor.markup.HighlighterLayer
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import org.evosuite.utils.CompactReport
+import java.awt.Color
+import kotlin.math.roundToInt
+
+class CoverageVisualisationService(private val project: Project) {
+
+    /**
+     * Shows coverage on the gutter next to the covered lines.
+     *
+     * @param testReport the generated tests summary
+     */
+    fun showCoverage(testReport: CompactReport) {
+        // Show toolWindow statistics
+        fillToolWindowContents(testReport)
+
+        // Show in-line coverage only if enabled in settings
+        val state = ApplicationManager.getApplication().getService(TestGenieSettingsService::class.java).state
+        if(state.showCoverage) {
+            val editor = FileEditorManager.getInstance(project).selectedTextEditor!!
+
+            val color = Color(100, 150, 20)
+
+            for (i in testReport.allCoveredLines) {
```

Comment:
What is the purpose of this for loop?

Reply:
It goes over each individual line that is covered in order to highlight it.